soci/soci_index.go

```diff
@@ -130,10 +202,11 @@ func GetIndexDescriptorCollection(ctx context.Context, cs content.Store, img ima
 }
 
 type buildConfig struct {
-	spanSize            int64
-	minLayerSize        int64
-	buildToolIdentifier string
-	platform            ocispec.Platform
+	spanSize              int64
+	minLayerSize          int64
+	buildToolIdentifier   string
+	platform              ocispec.Platform
+	supportLegacyRegistry bool
```

Comment:
I think we should use `legacyRegistry` than `supportLegacyRegistry`. `supportLegacyRegistry` indicates that, if false(not supported), then some functionality in soci might missing.

With this PR, we're supporting both `legacyRegistry` (false) and non-`legacyRegistry` (true). And `legacyRegistry` is just an indicator if we're using soci for a `legacyRegistry` or not.